### ISSUE TITLE
Consolidate react hooks under a single directory

### ISSUE THREAD

**Issue by allendav** [Automattic/woocommerce-payments#647]
After #617 lands, move use-confirm-navigation.js to the new hooks folder too

**Comment by zmaglica** [comment]
It looks like a technical debt issue that we should do either as a part of our cooldowns or as a hack week. Moving it from the engineering review queue since I think that we have enough context to tackle this issue.

This is the part of the Gamma backlog porter duty to reduce the number of the issues in the Engineering review column.

**Comment by naman03malhotra** [comment]
Looks like this issue can use a discussion. The question is do we really need this at this stage of WCPay or not. Probably something Front End guild can help with. @ismaeldcom

**Comment by ismaeldcom** [comment]
I agree that this could be a good issue for a cooldown. We don't have any urgency or need to consolidate our hooks.

That said, it might be nice to organize and document them. And maybe even look for some patterns that can take advantage of new ones.

**Comment by naman03malhotra** [comment]
Does the FE guild have a backlog(or list of issues) of its own? Maybe it can be added there, and someone can take it forward during the cooldown.

**Comment by ismaeldcom** [comment]
We don't have a backlog, for now our main focus is on TS. We can probably keep this issue adding the `cooldown` label.

**Comment by vbelolapotkov** [comment]
Closing this issue as stale. The codebase has changed a lot over the years, so if the problem is still relevant, I think its worth creating a new issue with updated focus and scope.
